enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,316 @@
+
+The backup sidecar container is being `disabled` by default. Upon creating a `Backup` CR on an Openshift cluster, the `PeriodicBackupController` reacts by enabling the container. This leads to a change in etcd static pod manifest, and a new revision is being rolled out. As explained above, such action causes etcd-cluster and API-Server unavailability.
+
+Additionally, any error within the backup sidecar, the container goes into `CrashLoopBackOff`, which leads to the whole `Pod` reports `NotReady` status. Consequently, the whole `Etcd-cluster` can not server requests, nor the `API-Server`.
+
+
+#### DaemonSet manages an Etcd-backup pod within every control-plane node.
+
+This approach separates the `etcd-backup-server` deployment from the etcd static pods, to avoid affecting etcd-cluster and API-Server availability upon any error within the etcd-backup-server.
+
+The advantage of using `DaemonSet` over Deployment is to guarantee scheduling etcd-backup-server pods on all `Control-plane` nodes, regardless of the number of nodes (e.g. 1, 3, 5) using `nodeSelector` capability.
+By setting the `nodeSelector` to the master node label (i.e. `node-role.kubernetes.io/master`), etcd-backup-server's pods are guaranteed to scheduled on every control-plane node.
+
+This method has been implemented and tested. The same container used in the Sidecar approach has been utilised. However, the scheduling is being managed by a DaemonSet.
+Upon creating a `Backup` CR on an Openshift cluster, the `PeriodicBackupController` reacts by creating a DaemonSet resource, with the `etcd-backup-server` container and the configurations applied in the CR.
+During [testing](https://github.com/openshift/cluster-etcd-operator/pull/1354#issuecomment-2408661261), this method has shown a solid and stable deployment, where the Backup pods shown only `1` restart over a duration of `45` minutes. Moreover, the cause of the restart on all the Backup pods were due to `node` issue, not the backup container.
+
+_**Pros**_
+
+As the Backup pods are being scheduled independently of the etcd pods, any error within the Backup pods does not influence cluster Openshift cluster availability.
+
+Moreover, as the Backup pods are being scheduled within every master nodes, the backups are being taken from every local etcd member on the same node. This avoids overwhelming a specific member with backup requests.
+
+Since a Backup pod exist on every master node, and a `hostPath` volume is being used to store the backups, it is guaranteed to have available backups on all master nodes, which is required for disaster recovery scenarios.
+
+_**Cons**_
+
+No issue has been encountered during testing this approach.
+
+### Open Questions
+
+- What values could be used for default configurations, considering all OCP variants.
+- What is the minimum disk size needed in order to operate this feature without running out of disk space ?
+- How to handle an empty `schedule` ? 
```

Comment:
I'd say empty schedule is invalid - its better be explicit than implicitly set "@weekly" or smth. Can set this to "weekly" default, but disallow unsetting it?

Reply:
alright, so QA found this bug https://issues.redhat.com/browse/OCPBUGS-43687 

There were two test scenarios 

- schedule: "" 
```
Fails with error:
The Backup "default" is invalid: spec.etcd.schedule: Invalid value: "": spec.etcd.schedule in body should match '^(@(annually|yearly|monthly|weekly|daily|hourly))|(\*|(?:\*|(?:[0-9]|(?:[1-5][0-9])))\/(?:[0-9]|(?:[1-5][0-9]))|(?:[0-9]|(?:[1-5][0-9]))(?:(?:\-[0-9]|\-
```
     
- schedule is being omitted completely from the CR 
```
$ oc logs backup-server-daemon-set-m96bv
F1022 13:26:10.239742       1 backupserver.go:77] error parsing backup schedule : empty spec string
```

I raised a fix for the second case, since the first is handled by the API-server 

https://github.com/openshift/cluster-etcd-operator/pull/1354/commits/564eb9ac3667ae9801c9276b3ac251f1d243be2b

Reply:
The fix uses default schedule which is 

```
defaultBackupSchedule = "0 0 * * *"
```

daily midnight